build-tools/packages/build-cli/src/commands/update/fluid-imports.ts

```diff
@@ -0,0 +1,332 @@
+		{ scope: "alpha", kind: "interface", name: "ISummaryNackMessage" },
+		{ scope: "alpha", kind: "interface", name: "ISummaryOpMessage" },
+		{ scope: "alpha", kind: "interface", name: "ISummaryRuntimeOptions" },
+		{ scope: "alpha", kind: "interface", name: "ISweepPhaseStats" },
+		{ scope: "alpha", kind: "interface", name: "IUploadSummaryResult" },
+		{ scope: "internal", kind: "const", name: "neverCancelledSummaryToken" },
+		{ scope: "alpha", kind: "type", name: "OpActionEventListener" },
+		{ scope: "alpha", kind: "type", name: "OpActionEventName" },
+		{
+			scope: "internal",
+			kind: "interface",
+			name: "RecentlyAddedContainerRuntimeMessageDetails",
+		},
+		{ scope: "internal", kind: "enum", name: "RuntimeHeaders" },
+		{ scope: "internal", kind: "enum", name: "RuntimeMessage" },
+		{ scope: "alpha", kind: "interface", name: "SubmitSummaryFailureData" },
+		{ scope: "alpha", kind: "type", name: "SubmitSummaryResult" },
+		{ scope: "alpha", kind: "class", name: "Summarizer" },
+		{ scope: "alpha", kind: "type", name: "SummarizeResultPart" },
+		{ scope: "alpha", kind: "type", name: "SummarizerStopReason" },
+		{ scope: "alpha", kind: "class", name: "SummaryCollection" },
+		{ scope: "alpha", kind: "type", name: "SummaryStage" },
+		{ scope: "alpha", kind: "const", name: "TombstoneResponseHeaderKey" },
+		{ scope: "internal", kind: "interface", name: "UnknownContainerRuntimeMessage" },
+		{ scope: "internal", kind: "function", name: "unpackRuntimeMessage" },
+	],
+	"@fluidframework/container-runtime-definitions": [
+		{ scope: "alpha", kind: "interface", name: "IContainerRuntime" },
+		{ scope: "alpha", kind: "type", name: "IContainerRuntimeBaseWithCombinedEvents" },
+		{ scope: "alpha", kind: "interface", name: "IContainerRuntimeEvents" },
+		{
+			scope: "alpha",
+			kind: "interface",
+			name: "IContainerRuntimeWithResolveHandle_Deprecated",
+		},
+	],
+};
+
+interface MemberData {
+	internal?: string[];
+	public?: string[];
+	alpha?: string[];
+	beta?: string[];
+}
+
+// Load the raw data into a more useable form
+type MapData = Map<string, MemberData>;
+const typeMapData: MapData = new Map();
+for (const [moduleName, members] of Object.entries(typeMapDataRaw)) {
+	const entry = typeMapData.get(moduleName) ?? {};
+	for (const member of members) {
+		switch (member.scope) {
+			case "internal": {
+				if (entry.internal === undefined) {
+					entry.internal = [member.name];
+				} else {
+					entry.internal.push(member.name);
+				}
+				break;
+			}
+			case "public": {
+				if (entry.public === undefined) {
+					entry.public = [member.name];
+				} else {
+					entry.public.push(member.name);
+				}
+				break;
+			}
+			case "alpha": {
+				if (entry.alpha === undefined) {
+					entry.alpha = [member.name];
+				} else {
+					entry.alpha.push(member.name);
+				}
+				break;
+			}
+			default: {
+				throw new Error(`Unknown API level: ${member.scope}`);
+			}
+		}
+	}
+	typeMapData.set(moduleName, entry);
+}
```

Comment:
This seems to be missing `beta`, but also shouldn't need to process each one-by-one.
Stick known in an array (see prior suggestion) and then see if unknown is in the list.
This isn't that useful when you have a canned list that is known to conform, but recommended arbitrary database earlier.
TypeScript has a limitation processing dynamic properties (doesn't maintain type assertions; so a helper is nice for that.)
I used `level` per `ApiLevel` rather than `scope` that `MemberDataRaw` has - probably should be made the same.

```suggestion
function isKnownLevel(level: string): level is ApiLevel {
	return (knownLevels as readonly string[]).includes(level);
}

function ensureLevel(entry: MemberData, level: keyof MemberData): string[] {
	const entryData = entry[level];
	if (entryData !== undefined) {
		return entryData;
	}
	const newData: string[] = [];
	entry[level] = newData;
	return newData;
}

// Load the raw data into a more useable form
const typeMapData = new Map<string, MemberData>();
type MapData = typeof typeMapData;
for (const [moduleName, members] of Object.entries(typeMapDataRaw)) {
	const entry = typeMapData.get(moduleName) ?? {};
	for (const member of members) {
		const { scope } = member;
		if (!isKnownLevel(scope)) {
			throw new Error(`Unknown API level: ${scope}`);
		}
		ensureLevel(entry, scope).push(member.name);
	}
	typeMapData.set(moduleName, entry);
}
```